specify/specify7
# Add basic tree creation functionality

Fixes #5268, as part of #4833 

This PR implements **Version 1** of the Tree Creation Process outlined in 4833. You should be able to create trees via the new add button in the Tree View. 🎉 

![image](https://github.com/user-attachments/assets/fb505f8b-dfa0-48ca-882a-5eddac68fb88)

This PR creates a tree with ranks, but without any items or anything--testing adding to the new tree/etc isn't apart of testing since it doesn't come with a root node by default!

### Checklist

      and self-explanatory (or properly documented)

### Testing instructions




- Log in as an admin.
- [ ] Go to the Storage and Geography tree views and make sure that the Add button is _not_ present on either tree.
- - Add button will appear after the edit button next to the tree's name.
- [ ] Go to the Taxon tree view and see that the Add button is there.
- [ ] After clicking on the Add button, ensure that a dialog with a list of default trees appear. Make sure you can select each one.
- - The list should include Botany, Entomology, Herpetology, Ichthyology, Invertebrate Paleontology, Invertebrate Zoology, Mammalogy, Ornithology, Paleobotany, Vascular Plants, and Vertebrate Paleontology
- [ ] After selecting a record, make sure that you **cannot add any TreeDefItems**, but can still edit and delete existing ones.
- - Check this on multiple subview types--subform, grid, and button. Button is the default, and the XML for a diff form is below.
- [ ] Upon saving, make sure that the new tree is in the tree dropdown.

#### Permissions
- Log in as a non-admin.
- [ ] Go to the Taxon tree view and make sure that the Add button is _not_ there.

Here is the taxon tree definition form used in development, from #5073 (thanks Jason!)--it'll be added as the default form in `v7.9.8`, but this PR should work with any form. You'll need to click 'Show all tables' in App Resources--also make sure that you're adding it to Taxon Tree Definition, and not Taxon Tree Definition Item.

```xml
<viewdef name="TaxonTreeDef" class="edu.ku.brc.specify.datamodel.TaxonTreeDef" type="form" gettable="edu.ku.brc.af.ui.forms.DataGetterForObj" settable="edu.ku.brc.af.ui.forms.DataSetterForObj">
	<desc>The TaxonTreeDef Table</desc>
	<enableRules/>
	<columnDef>p,2px,p,2px,p,2px,p,2px,p,2px,p,p:g</columnDef>
	<rowDef auto="true" cell="p" sep="2px"/>
	<rows>
		<row>
			<cell type="label" labelfor="name"/>
			<cell type="field" id="name" name="name" uitype="text"/>
		</row>
		<row>
			<cell type="label" labelfor="fullNameDir"/>
			
			<cell type="field" id="fullNameDir" name="fullNameDirection" default="1" uitype="combobox"/>
		</row>
		<row>
			<cell type="label" labelfor="remarks"/>
			<cell type="field" id="remarks" name="remarks" uitype="textareabrief" rows="2" colspan="6"/>
		</row>
		<row>
                         
			<cell type="subview" id="items" name="treeDefItems" colspan="8" defaulttype="table" initialize="sortField=rankId"/>
		</row>
	</rows>
</viewdef>
```
